docs/project/contribution_tools.md

```diff
@@ -107,13 +107,10 @@ brew install \
   bazelisk \
   gh \
   llvm \
-  python@3.10
```

Comment:
Carbon has a number of Python scripts, and part of the intent here was to ensure there's a good version to use. I think it's fine to switch pre-commit to a brew install, but this should probably stay.

Reply:
Okay. That makes sense. I had excluded Python because it's a transitive dependency of pre-commit, but it's probably better to be explicit.

Should we leave it at Python 3.10, or up it to 3.12 (which is what pre-commit depends on)?

Reply:
Generally we've been saying we require 3.9 or newer (lower on the page), which is where this came from. I think part of the desire for an older version here stems from wanting to stop people from depending on a feature specific to a newer version of python, breaking compatibility. It's a delicate balance. :-/